### PR TITLE
Switch to importlib.metadata to list entry points

### PR DESCRIPTION
**Pull request by fcollonval**
Alignment following https://github.com/jupyterlab/jupyterlab/pull/12293

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -36,7 +36,7 @@ packages = find:
 python_requires = >=3.7
 install_requires =
     babel
-    entrypoints>=0.2.2
+    importlib_metadata>=1.4;python_version=="3.7"
```
Comment (bollwyvl) [inline setup.cfg:39]: unfortunately, the situation isn't quite this simple, as the `group` argument wasn't introduced until [3.10](https://docs.python.org/3.10/library/importlib.metadata.html#entry-points) and wasn't added until 3.6 of [importlib-metadata](https://importlib-metadata.readthedocs.io/en/latest/using.html#entry-points).

So unfortunately, the safest  thing to do at this point is more like:

```ini
install_requires =
    importlib_metadata>=3.6;python_version<3.10
```

and then

```py
import sys

if sys.version_info < (3,10):
    from importlib_metadata import entry_points 
else:
    from importlib.metadata import entry_points 

# ...
    entry_points(group=JUPYTERLAB_LANGUAGEPACK_ENTRY)
```

Comment (fcollonval) [reply setup.cfg]: Thanks @bollwyvl

File: jupyterlab_server/translation_utils.py [hunk 2]
```diff
@@ -14,9 +14,13 @@
 from typing import Dict, Pattern, Tuple
 
 import babel
-import entrypoints
 from packaging.version import parse as parse_version
 
+if sys.version_info < (3, 10):
```
Comment (bollwyvl) [inline jupyterlab_server/translation_utils.py:19]: Might even be worth linking to that thing about why this complexity is required.

**Review by blink1073 — APPROVED** [review]
Thanks!

File: jupyterlab_server/translation_utils.py (hunk 2 above)
Comment (fcollonval) [reply jupyterlab_server/translation_utils.py:20]: Thanks

**Pull request merged**